woudc_data_registry/models.py

```diff
@@ -1163,12 +1250,113 @@ def sync(ctx):
         Station,
         Instrument,
         Deployment,
-        DataRecord
+        DataRecord,
     ]
+    contribution_models = list()
+
+    """function to form all the contributions from the other models lists"""
+    def build_contributions(contribution_models, project_models,
+                            station_models, instrument_models, country_models,
+                            deployment_models, contributor_models):
+
+        """turn lists holding essential information into dictionaries with key
+        that is common with another list in order to allow for fast lookup"""
+        country_dict = {country_models[i].country_id: country_models[i]
+                        for i in range(0, len(country_models))}
+        instrument_dict = {instrument_models[i].name: instrument_models[i]
+                           for i in range(0, len(instrument_models))}
+        contributor_dict = {contributor_models[i].contributor_id:
+                            contributor_models[i]
+                            for i in range(0, len(contributor_models))}
+
+        """deployment_dict will have stations as
+        keys in order to help connect them to projects """
+        deployment_dict = {deployment_models[i].station_id:
+                           deployment_models[i]
+                           for i in range(0, len(deployment_models))}
```

Comment:
Be careful of how dictionaries can overwrite a previous value when you try to insert something with the same key.
There are often multiple deployments with the same station, and here only one of those will end up in the final `dict`.

One solution would be to replace the `station_id` in the `Instrument` class with a `deployment_id`, we would need to cooperate to do that but it's possible and would be helpful in other places too.